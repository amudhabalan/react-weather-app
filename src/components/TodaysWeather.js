import React from "react";
import Grid from "@material-ui/core/Grid";
import Moment from "react-moment";

const TodaysWeather = ({ weather }) => (
  <React.Fragment>
    <Grid item xs={6}>
      <p>
        <Moment date={weather.applicable_date} format="dddd MMM, Do" />
        <br></br>
        {weather.weather_state_name}
        <br></br>
        <img
          src={
            "https://www.metaweather.com/static/img/weather/ico/" +
            weather.weather_state_abbr +
            ".ico"
          }
          alt={weather.weather_state_name}
        ></img><h1>{Math.round(weather.the_temp)}<span>&#8451;</span></h1>
      </p>
    </Grid>
    <Grid item xs={6}>
      <p>
        Wind Speed: {Math.round(weather.wind_speed)} mph
        <br></br>Humidity: {Math.round(weather.humidity)}%<br></br>Air Pressure:{" "}
        {Math.round(weather.air_pressure)} mbar
      </p>
    </Grid>
  </React.Fragment>
);

export default TodaysWeather;
