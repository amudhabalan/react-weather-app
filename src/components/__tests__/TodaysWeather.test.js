import React from 'react';
import TodaysWeather from '../TodaysWeather';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

configure({ adapter: new Adapter() });

const weather = {
  dt: 1590387700,
  sunrise: 1590330189,
  sunset: 1590382241,
  temp: 16.98,
  feels_like: 14.62,
  pressure: 1021,
  humidity: 93,
  dew_point: 15.84,
  uvi: 9.92,
  clouds: 100,
  wind_speed: 6.12,
  wind_deg: 221,
  weather: [
    {
      id: 804,
      main: 'Clouds',
      description: 'overcast clouds',
      icon: '04n'
    }
  ]
};

describe('TodaysWeather', () => {
  it('should Match Snapshot', () => {
    const wrapper = shallow(
      <TodaysWeather city="Melbourne" weather={weather} max="17" min="13" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
