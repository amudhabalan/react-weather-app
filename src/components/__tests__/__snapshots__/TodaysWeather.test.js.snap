// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodaysWeather should Match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodaysWeather
    city="Melbourne"
    max="17"
    min="13"
    weather={
        Object {
            "clouds": 100,
            "dew_point": 15.84,
            "dt": 1590387700,
            "feels_like": 14.62,
            "humidity": 93,
            "pressure": 1021,
            "sunrise": 1590330189,
            "sunset": 1590382241,
            "temp": 16.98,
            "uvi": 9.92,
            "weather": Array [
              Object {
                "description": "overcast clouds",
                "icon": "04n",
                "id": 804,
                "main": "Clouds",
              },
            ],
            "wind_deg": 221,
            "wind_speed": 6.12,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="todays-weather-left"
>
          <h1>
                    Melbourne
          </h1>
          <p>
                    <t
                              ago={false}
                              calendar={false}
                              date={1590387700}
                              decimal={false}
                              element={null}
                              filter={[Function]}
                              format="dddd MMM, Do"
                              fromNow={false}
                              interval={60000}
                              local={false}
                              onChange={[Function]}
                              titleFormat=""
                              toNow={false}
                              unit={null}
                              unix={true}
                              utc={false}
                              withTitle={false}
                    />
          </p>
          <h2>
                    17
                    <span>
                              ℃
                    </span>
                     
                    <img
                              alt="overcast clouds"
                              src="http://openweathermap.org/img/wn/04n@2x.png"
                    />
          </h2>
          <p>
                    Clouds
          </p>
</div>,
        <div
          className="todays-weather-right"
>
          <div
                    className="mode-switch"
          >
                    <label
                              className="switch"
                    >
                              <input
                                        onClick={undefined}
                                        type="checkbox"
                              />
                              <span
                                        className="slider round"
                              />
                    </label>
          </div>
          <div
                    className="weather-details"
          >
                    <p>
                              Max: 
                              17
                              <span>
                                        ℃
                              </span>
                    </p>
                    <p>
                              Min: 
                              13
                              <span>
                                        ℃
                              </span>
                    </p>
                    <p>
                              Wind Speed: 
                              6
                               mph
                    </p>
                    <p>
                              Humidity: 
                              93
                              %
                    </p>
                    <p>
                              Pressure: 
                              1021
                               mbar
                    </p>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Melbourne
</h1>,
            <p>
              <t
                            ago={false}
                            calendar={false}
                            date={1590387700}
                            decimal={false}
                            element={null}
                            filter={[Function]}
                            format="dddd MMM, Do"
                            fromNow={false}
                            interval={60000}
                            local={false}
                            onChange={[Function]}
                            titleFormat=""
                            toNow={false}
                            unit={null}
                            unix={true}
                            utc={false}
                            withTitle={false}
              />
</p>,
            <h2>
              17
              <span>
                            ℃
              </span>
               
              <img
                            alt="overcast clouds"
                            src="http://openweathermap.org/img/wn/04n@2x.png"
              />
</h2>,
            <p>
              Clouds
</p>,
          ],
          "className": "todays-weather-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Melbourne",
            },
            "ref": null,
            "rendered": "Melbourne",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <t
                ago={false}
                calendar={false}
                date={1590387700}
                decimal={false}
                element={null}
                filter={[Function]}
                format="dddd MMM, Do"
                fromNow={false}
                interval={60000}
                local={false}
                onChange={[Function]}
                titleFormat=""
                toNow={false}
                unit={null}
                unix={true}
                utc={false}
                withTitle={false}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "ago": false,
                "calendar": false,
                "date": 1590387700,
                "decimal": false,
                "element": null,
                "filter": [Function],
                "format": "dddd MMM, Do",
                "fromNow": false,
                "interval": 60000,
                "local": false,
                "onChange": [Function],
                "titleFormat": "",
                "toNow": false,
                "unit": null,
                "unix": true,
                "utc": false,
                "withTitle": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                17,
                <span>
                  ℃
</span>,
                " ",
                <img
                  alt="overcast clouds"
                  src="http://openweathermap.org/img/wn/04n@2x.png"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              17,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "℃",
                },
                "ref": null,
                "rendered": "℃",
                "type": "span",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "overcast clouds",
                  "src": "http://openweathermap.org/img/wn/04n@2x.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Clouds",
            },
            "ref": null,
            "rendered": "Clouds",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mode-switch"
>
              <label
                            className="switch"
              >
                            <input
                                          onClick={undefined}
                                          type="checkbox"
                            />
                            <span
                                          className="slider round"
                            />
              </label>
</div>,
            <div
              className="weather-details"
>
              <p>
                            Max: 
                            17
                            <span>
                                          ℃
                            </span>
              </p>
              <p>
                            Min: 
                            13
                            <span>
                                          ℃
                            </span>
              </p>
              <p>
                            Wind Speed: 
                            6
                             mph
              </p>
              <p>
                            Humidity: 
                            93
                            %
              </p>
              <p>
                            Pressure: 
                            1021
                             mbar
              </p>
</div>,
          ],
          "className": "todays-weather-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <label
                className="switch"
>
                <input
                                onClick={undefined}
                                type="checkbox"
                />
                <span
                                className="slider round"
                />
</label>,
              "className": "mode-switch",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    onClick={undefined}
                    type="checkbox"
/>,
                  <span
                    className="slider round"
/>,
                ],
                "className": "switch",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": undefined,
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "slider round",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Max: 
                  17
                  <span>
                                    ℃
                  </span>
</p>,
                <p>
                  Min: 
                  13
                  <span>
                                    ℃
                  </span>
</p>,
                <p>
                  Wind Speed: 
                  6
                   mph
</p>,
                <p>
                  Humidity: 
                  93
                  %
</p>,
                <p>
                  Pressure: 
                  1021
                   mbar
</p>,
              ],
              "className": "weather-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Max: ",
                    17,
                    <span>
                      ℃
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Max: ",
                  17,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "℃",
                    },
                    "ref": null,
                    "rendered": "℃",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Min: ",
                    13,
                    <span>
                      ℃
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Min: ",
                  13,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "℃",
                    },
                    "ref": null,
                    "rendered": "℃",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Wind Speed: ",
                    6,
                    " mph",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Wind Speed: ",
                  6,
                  " mph",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Humidity: ",
                    93,
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Humidity: ",
                  93,
                  "%",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Pressure: ",
                    1021,
                    " mbar",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Pressure: ",
                  1021,
                  " mbar",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="todays-weather-left"
>
            <h1>
                        Melbourne
            </h1>
            <p>
                        <t
                                    ago={false}
                                    calendar={false}
                                    date={1590387700}
                                    decimal={false}
                                    element={null}
                                    filter={[Function]}
                                    format="dddd MMM, Do"
                                    fromNow={false}
                                    interval={60000}
                                    local={false}
                                    onChange={[Function]}
                                    titleFormat=""
                                    toNow={false}
                                    unit={null}
                                    unix={true}
                                    utc={false}
                                    withTitle={false}
                        />
            </p>
            <h2>
                        17
                        <span>
                                    ℃
                        </span>
                         
                        <img
                                    alt="overcast clouds"
                                    src="http://openweathermap.org/img/wn/04n@2x.png"
                        />
            </h2>
            <p>
                        Clouds
            </p>
</div>,
          <div
            className="todays-weather-right"
>
            <div
                        className="mode-switch"
            >
                        <label
                                    className="switch"
                        >
                                    <input
                                                onClick={undefined}
                                                type="checkbox"
                                    />
                                    <span
                                                className="slider round"
                                    />
                        </label>
            </div>
            <div
                        className="weather-details"
            >
                        <p>
                                    Max: 
                                    17
                                    <span>
                                                ℃
                                    </span>
                        </p>
                        <p>
                                    Min: 
                                    13
                                    <span>
                                                ℃
                                    </span>
                        </p>
                        <p>
                                    Wind Speed: 
                                    6
                                     mph
                        </p>
                        <p>
                                    Humidity: 
                                    93
                                    %
                        </p>
                        <p>
                                    Pressure: 
                                    1021
                                     mbar
                        </p>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Melbourne
</h1>,
              <p>
                <t
                                ago={false}
                                calendar={false}
                                date={1590387700}
                                decimal={false}
                                element={null}
                                filter={[Function]}
                                format="dddd MMM, Do"
                                fromNow={false}
                                interval={60000}
                                local={false}
                                onChange={[Function]}
                                titleFormat=""
                                toNow={false}
                                unit={null}
                                unix={true}
                                utc={false}
                                withTitle={false}
                />
</p>,
              <h2>
                17
                <span>
                                ℃
                </span>
                 
                <img
                                alt="overcast clouds"
                                src="http://openweathermap.org/img/wn/04n@2x.png"
                />
</h2>,
              <p>
                Clouds
</p>,
            ],
            "className": "todays-weather-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Melbourne",
              },
              "ref": null,
              "rendered": "Melbourne",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <t
                  ago={false}
                  calendar={false}
                  date={1590387700}
                  decimal={false}
                  element={null}
                  filter={[Function]}
                  format="dddd MMM, Do"
                  fromNow={false}
                  interval={60000}
                  local={false}
                  onChange={[Function]}
                  titleFormat=""
                  toNow={false}
                  unit={null}
                  unix={true}
                  utc={false}
                  withTitle={false}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "ago": false,
                  "calendar": false,
                  "date": 1590387700,
                  "decimal": false,
                  "element": null,
                  "filter": [Function],
                  "format": "dddd MMM, Do",
                  "fromNow": false,
                  "interval": 60000,
                  "local": false,
                  "onChange": [Function],
                  "titleFormat": "",
                  "toNow": false,
                  "unit": null,
                  "unix": true,
                  "utc": false,
                  "withTitle": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  17,
                  <span>
                    ℃
</span>,
                  " ",
                  <img
                    alt="overcast clouds"
                    src="http://openweathermap.org/img/wn/04n@2x.png"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                17,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "℃",
                  },
                  "ref": null,
                  "rendered": "℃",
                  "type": "span",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "overcast clouds",
                    "src": "http://openweathermap.org/img/wn/04n@2x.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Clouds",
              },
              "ref": null,
              "rendered": "Clouds",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mode-switch"
>
                <label
                                className="switch"
                >
                                <input
                                                onClick={undefined}
                                                type="checkbox"
                                />
                                <span
                                                className="slider round"
                                />
                </label>
</div>,
              <div
                className="weather-details"
>
                <p>
                                Max: 
                                17
                                <span>
                                                ℃
                                </span>
                </p>
                <p>
                                Min: 
                                13
                                <span>
                                                ℃
                                </span>
                </p>
                <p>
                                Wind Speed: 
                                6
                                 mph
                </p>
                <p>
                                Humidity: 
                                93
                                %
                </p>
                <p>
                                Pressure: 
                                1021
                                 mbar
                </p>
</div>,
            ],
            "className": "todays-weather-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <label
                  className="switch"
>
                  <input
                                    onClick={undefined}
                                    type="checkbox"
                  />
                  <span
                                    className="slider round"
                  />
</label>,
                "className": "mode-switch",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      onClick={undefined}
                      type="checkbox"
/>,
                    <span
                      className="slider round"
/>,
                  ],
                  "className": "switch",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "onClick": undefined,
                      "type": "checkbox",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "slider round",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "label",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Max: 
                    17
                    <span>
                                        ℃
                    </span>
</p>,
                  <p>
                    Min: 
                    13
                    <span>
                                        ℃
                    </span>
</p>,
                  <p>
                    Wind Speed: 
                    6
                     mph
</p>,
                  <p>
                    Humidity: 
                    93
                    %
</p>,
                  <p>
                    Pressure: 
                    1021
                     mbar
</p>,
                ],
                "className": "weather-details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Max: ",
                      17,
                      <span>
                        ℃
</span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Max: ",
                    17,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "℃",
                      },
                      "ref": null,
                      "rendered": "℃",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Min: ",
                      13,
                      <span>
                        ℃
</span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Min: ",
                    13,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "℃",
                      },
                      "ref": null,
                      "rendered": "℃",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Wind Speed: ",
                      6,
                      " mph",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Wind Speed: ",
                    6,
                    " mph",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Humidity: ",
                      93,
                      "%",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Humidity: ",
                    93,
                    "%",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Pressure: ",
                      1021,
                      " mbar",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Pressure: ",
                    1021,
                    " mbar",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
